views/default/contests.html

```diff
@@ -131,19 +138,78 @@
                 });
 
-                request.execute(function(event) {
+                request.execute(function (event) {
                     $.web2py.flash("Reminder added!");
+                    checkContests();
                 });
             }
 
             /* Call Auth for Google Calendar */
-            handleAuthClick();
+            handleAuthorize(addEventToCalendar);
+        }
+
+        /* check contests using link and data/time */
+        function checkContests() {
+            $('#contests-table').find('> tbody > tr')
+                    /* only ongoing contests*/
+                    .filter(function (index, element) {
+                        return this.children.length === 6 && $('.stopstalk-timestamp', this).length === 1;
+                    })
+                    /* refer to https://developers.google.com/google-apps/calendar/v3/reference/events/list
+                     * for more info about constructing listing request
+                     */
+                    .each(function () {
+
+                        var contestName = this.children[0].textContent;
+                        var siteName = this.children[1].textContent;
+                        var startTime = this.children[2].textContent;
+                        // Note this will always be duration and not endtime
+                        var duration = this.children[3].textContent;
+                        var contestLink = this.children[4].firstChild.href;
+                        var calendarButton = $('.set-reminder', this);
+
+                        var startTimeStamp = new Date(startTime.replace(/-/g, '/'));
+                        var endTimeStamp = dateToTimeStamp(addMinutes(startTimeStamp, getMinutes(duration) - 5));
+                        startTimeStamp = dateToTimeStamp(addMinutes(startTimeStamp, -5));
+
+                        var request = gapi.client.calendar.events.list({
+                            'calendarId': 'primary',
+                            'maxResults': 20,
+                            'singleEvents': 'true',
+                            'orderBy': 'startTime',
+                            'q': contestLink,
+                            'timeMin': startTimeStamp,
+                            'timeMax': endTimeStamp
+                        });
+                        /* check contest existence */
+                        request.execute(function (resp) {
+                            var events = resp.items;
+                            calendarButton.removeClass('disabled');
+                            calendarButton.attr('data-tooltip', 'Set Reminder to Google Calendar');
+                            $.each(events, function (index, element) {
+                                if (this.summary === 'Contest at ' + siteName && this.location === contestLink) {
+                                    calendarButton.addClass('disabled');
+                                    calendarButton.attr('data-tooltip', 'Already set!');
```

Comment:
Now this leads to a bit of confusion among the disabled buttons.
Please disable the buttons and add a tick icon for the added contests - `fa-check`
But I would be open for other awesome suggestions for this button.